### PR TITLE
Some CI workflow suggestions.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Apologies that this is a bit of a grab-bag PR. Some annocations on the proposed changes will be added as line comments.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -2,11 +2,12 @@ name: Deploy Website to Github Pages
...
 on:
   push:
-    branches: [ master ]
+    branches: [master]
+  pull_request:
 
 jobs:
   build_and_deploy:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (nuclearsandwich) [inline .github/workflows/deploy.yml:10]: In general I prefer and recommend using explicit host platform versions, it helps prevent random breakages when they move the "latest" platform underneath you. In this case we are downloading a specific dependency which is only built for 20.04 so we should also use that as our build platform.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -34,6 +35,7 @@ jobs:
     # Deploy the site built into the gh-pages branch.
     # You need to configure pages to be at gh-pages and serve the root folder.
     - name: Deploy
+      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
```
Comment (nuclearsandwich) [inline .github/workflows/deploy.yml:38]: This conditional allows us to run builds on each PR, validating the site builds, while still only deploying when the push to master goes through.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (german-e-mas) [reply .github/workflows/deploy.yml:10]: Thanks! That's a great point.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -2,11 +2,12 @@ name: Deploy Website to Github Pages
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
+  pull_request:
```
Comment (german-e-mas) [inline .github/workflows/deploy.yml:6]: So this means it will trigger for each pull request made, right?

**Review by german-e-mas — APPROVED** [review]
LGTM!

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (nuclearsandwich) [reply .github/workflows/deploy.yml:6]: correct! You can see that it actually triggered for this PR (and that the deploy step was skipped).

**Pull request merged**